moveit_ros/benchmarks/src/BenchmarkOptions.cpp

```diff
@@ -260,40 +271,109 @@ void BenchmarkOptions::readWorkspaceParameters(const rclcpp::Node::SharedPtr& no
 }
 
-void BenchmarkOptions::readPlannerConfigs(const rclcpp::Node::SharedPtr& node)
+bool BenchmarkOptions::readPlannerConfigs(const rclcpp::Node::SharedPtr& node)
 {
+  const std::string ns = "benchmark_config.planning_pipelines";
+  // pipelines
   planning_pipelines_.clear();
 
-  const std::string np = "benchmark_config.planning_pipelines";
   std::vector<std::string> pipelines;
-  if (!node->get_parameter(np + ".pipelines", pipelines))
+  if (!node->get_parameter(ns + ".pipelines", pipelines))
   {
-    RCLCPP_ERROR(LOGGER, "Fail to get the parameter in `%s` namespace.", (np + ".pipelines").c_str());
-    return;
+    RCLCPP_ERROR(LOGGER, "Fail to get the parameter in `%s` namespace.", (ns + ".pipelines").c_str());
+    return false;
   }
 
   for (const std::string& pipeline : pipelines)
   {
-    std::string pipeline_name;
-    const std::string pipeline_parameter_name = std::string(np).append(".").append(pipeline).append(".name");
-    if (!node->get_parameter(pipeline_parameter_name, pipeline_name))
+    if (!pipeline.empty())
     {
-      RCLCPP_ERROR(LOGGER, "Fail to get the parameter in `%s` namespace.", pipeline_parameter_name.c_str());
-      return;
+      std::string pipeline_name;
+      const std::string pipeline_parameter_name = std::string(ns).append(".").append(pipeline).append(".name");
+      if (!node->get_parameter(pipeline_parameter_name, pipeline_name))
+      {
+        RCLCPP_ERROR(LOGGER, "Fail to get the parameter in `%s` namespace.", pipeline_parameter_name.c_str());
+        return false;
+      }
+
+      RCLCPP_INFO(LOGGER, "Reading in planner names for planning pipeline '%s'", pipeline_name.c_str());
+
+      std::vector<std::string> planners;
+      const std::string pipeline_parameter_planners = std::string(ns).append(".").append(pipeline).append(".planners");
+      if (!node->get_parameter(pipeline_parameter_planners, planners))
+      {
+        RCLCPP_ERROR(LOGGER, "Fail to get the parameter in `%s` namespace.", pipeline_parameter_planners.c_str());
+        return false;
+      }
+
+      for (const std::string& planner : planners)
+      {
+        RCLCPP_INFO(LOGGER, "  %s", planner.c_str());
+      }
+
+      planning_pipelines_[pipeline_name] = planners;
     }
+  }
+  // parallel pipelines
+  parallel_planning_pipelines_.clear();
 
-    RCLCPP_INFO(LOGGER, "Reading in planner names for planning pipeline '%s'", pipeline_name.c_str());
+  std::vector<std::string> parallel_pipelines;
+  if (!node->get_parameter(ns + ".parallel_pipelines", parallel_pipelines))
+  {
+    RCLCPP_ERROR(LOGGER, "Fail to get the parameter in `%s` namespace.", (ns + ".parallel_pipelines").c_str());
+    return false;
+  }
 
-    std::vector<std::string> planners;
-    if (!node->get_parameter(pipeline_parameter_name, planners))
-    {
-      RCLCPP_ERROR(LOGGER, "Fail to get the parameter in `%s` namespace.", pipeline_parameter_name.c_str());
-      return;
+  for (const std::string& parallel_pipeline : parallel_pipelines)
```

Comment:
Wouldn't it be simpler to just use the pipelines that are configured with MoveItCpp, and only reference the to-be-benchmarked pipeline ids?

Reply:
Not sure if I understand your proposal correctly but I think it is more useful to give both pieces of information for example if you want to use the same planner_id in two different pipelines that have a different set of plan_request_adapters or trajectory generation algorithms. For example, two OMPL pipelines that both use RRTConnect but one uses ruckig and the other one just TOTG